### DISCUSSION TITLE
Make disable clearance checks a local setting

### DISCUSSION THREAD

**Discussion by Margen67** [OpenRCT2/OpenRCT2#15692] [Feature requests]
Since having clearance checks disabled can be quite annoying and buggy ( #2053, #1613 ) how about we let players re-enable clearance checks while the cheat is on?

**Comment by Ziscor** [comment]
I think it's best we keep this permission-locked from both ways as it is. Even if some players might find it hard to work with, someone probably turned on the cheat for a reason, you know. Realistic Servers find this very useful and enable this cheat often, but if someone were to disable it without fore-warning all players, wouldn't _that_ be annoying to the persons who were currently using it too? 

You can always ask the Host/ Admin/ Moderator to turn the cheat off. I'm sure given that everyone is done using it, they'll turn it off soon enough.

**Comment by IntelOrca** [comment]
The cheat should be allowing clients to use disable clearance. The cheat shouldn't actually enable it for every player - only the ability (it should be local). That is what needs resolving, nothing else.

A bit awkward to do this at the moment as we don't pass whether checks are disabled via the game commands.

**Reply by fidwell** [reply]
Hey it's been a few years; is this any more feasible now than it used to be? As someone who hosts multiplayer a lot, it can be frustrating when clearance checks is a global setting instead of per-user.

**Reply by RayKoopa** [reply]
I'd really enjoy a per-user CC if the server gives permission to enable it as it's currently a hickhack between users constantly enabling/disabling it. There are even some warnings made out of decoration on FFA maps to turn it off before leaving the server to not confuse players joining in afterwards.

Thinking a bit more about it, maybe all of these should then be a user-specific aswell as they only change what is possible through actions after toggling them, not affecting game state immediately:
- Enable sandbox mode
- Disable clearance checks
- Disable support limits
- Allow building in pause mode (kinda unnecessary as multiplayer cannot be paused AFAIK)
- Allow using regular paths as queue
- Allow special color schemes
- Likely all "Ride cheats" tab checkboxes

**Comment by Margen67** [comment]
Maybe you could have the check over the cheat be translucent to show that you can activate the cheat while it's disabled?

**Comment by Nubbie** [comment]
Would rather want to get #3555 or like IntelOrca said, having it locally

**Comment by Ryder17z** [comment]
I prefer #3555

**Comment by Margen67** [comment]
I think it would be best to have both.
